modules/basegl/processors/splitimage.cpp

```diff
@@ -0,0 +1,145 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/splitimage.h>
+
+#include <modules/opengl/openglutils.h>
+#include <modules/opengl/sharedopenglresources.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/shader/shader.h>
+#include <modules/opengl/image/imagegl.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo SplitImage::processorInfo_{
+    "org.inviwo.SplitImage",  // Class identifier
+    "Split Image",            // Display name
+    "Image Operation",        // Category
+    CodeState::Stable,        // Code state
+    Tags::GL,                 // Tags
+};
+const ProcessorInfo SplitImage::getProcessorInfo() const { return processorInfo_; }
+
+SplitImage::SplitImage()
+    : Processor()
+    , inport0_("inputA")
+    , inport1_("inputB")
+    , outport_("outport")
+    , splitDirection_("splitDirection", "Split Direction",
+                      {{"vertical", "Vertical", SplitDirection::Vertical},
+                       {"horizontal", "Horizontal", SplitDirection::Horizontal}},
+                      0)
+    , splitPosition_("splitPosition", "Split Position", 0.5f, 0.0f, 1.0f) {
+
+    inport0_.setOptional(true);
+    inport1_.setOptional(true);
+    addPort(inport0_);
+    addPort(inport1_);
+    addPort(outport_);
+
+    addProperty(splitDirection_);
+    addProperty(splitPosition_);
+}
+
+void SplitImage::process() {
+    utilgl::activateAndClearTarget(outport_);
+
+    if (!inport0_.hasData() && !inport1_.hasData()) {
+        // no inputs, render noise
+
+        auto shader = SharedOpenGLResources::getPtr()->getNoiseShader();
+        shader->activate();
+        utilgl::singleDrawImagePlaneRect();
+        shader->deactivate();
+    } else {
+        utilgl::DepthMaskState depthMask(GL_FALSE);
+        utilgl::GlBoolState depth(GL_DEPTH_TEST, GL_FALSE);
+        utilgl::GlBoolState scissor(GL_SCISSOR_TEST, GL_TRUE);
+
+        const ivec2 dims(outport_.getDimensions());
+
+        ivec4 viewport0;
+        ivec4 viewport1;
+        if (splitDirection_.get() == SplitDirection::Vertical) {
```

Comment:
ImageLayoutGL has logic for splitting as well. 
Could perhaps be adopted in future PR...